### PR TITLE
feat(pci.data-processing): dashboard - hide graphs if no data

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-7

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/datatr-3-dataprocessing-jobs-dashboard f`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-7
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Hide the graphs in the dashboard if there is no data.

## Related

```
DATATR-3
  ^ 
  | - DATATR-6
  | - DATATR-7
  | - DATATR-10
  | - DATATR-11
  | - DATATR-12
  | - DATATR-13
```
[DATATR-3](https://github.com/ovh/manager/tree/feat/datatr-3-dataprocessing-jobs-dashboard)
[DATATR-6](https://github.com/ovh/manager/pull/7120)
[DATATR-7](https://github.com/ovh/manager/pull/7115)
[DATATR-10](https://github.com/ovh/manager/pull/7116)
[DATATR-11](https://github.com/ovh/manager/pull/7119)
[DATATR-12](https://github.com/ovh/manager/pull/7117)
[DATATR-13](https://github.com/ovh/manager/pull/7118)

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 1]
```diff
@@ -33,61 +33,70 @@
...
-                class="information-tile__definition"
-                data-translate="data_processing_details_charts_total_memory"
-            ></span>
-            <div
-                class="chart__nodata-container"
-                ng-if="$ctrl.metrics.totalMemory.data.length === 0"
-            >
-                <span data-translate="data_processing_details_no_data"></span>
+            <!--Charts -->
+            <div ng-if="$ctrl.hasCharts()">
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:37]:
```suggestion
            <div data-ng-if="$ctrl.hasCharts()">
```

Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html]: Done here: https://github.com/ovh/manager/pull/7115/commits/4b20dd8f50df52ca1392228f89c00975e2a5f6c0

**Review by Dimfacion — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 2]
```diff
@@ -33,61 +33,70 @@
...
+            <!--Charts -->
+            <div data-ng-if="$ctrl.hasCharts()">
+                <!-- Tile item :: Total memory -->
+                <span
+                    class="information-tile__definition"
+                    data-translate="data_processing_details_charts_total_memory"
+                ></span>
+                <div
+                    class="chart__nodata-container"
+                    ng-if="$ctrl.metrics.totalMemory.data.length === 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:45]:
```suggestion
                    data-ng-if="$ctrl.metrics.totalMemory.data.length === 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 3]
```diff
@@ -33,61 +33,70 @@
...
+                >
+                    <span
+                        data-translate="data_processing_details_no_data"
+                    ></span>
+                </div>
+                <pci-project-data-processing-metrics-chart
+                    ng-if="$ctrl.metrics.totalMemory.data.length !== 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:52]:
```suggestion
                    data-ng-if="$ctrl.metrics.totalMemory.data.length !== 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 4]
```diff
@@ -33,61 +33,70 @@
...
+                >
+                </pci-project-data-processing-metrics-chart>
+                <!-- Tile item :: Active tasks -->
+                <span class="information-tile__definition"
+                    >{{ :: 'data_processing_details_charts_num_active_tasks' |
+                    translate }}</span
+                >
+                <div
+                    class="chart__nodata-container"
+                    ng-if="$ctrl.metrics.activeTasks.data.length === 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:66]:
```suggestion
                    data-ng-if="$ctrl.metrics.activeTasks.data.length === 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 5]
```diff
@@ -33,61 +33,70 @@
...
+                >
+                    <span
+                        data-translate="data_processing_details_no_data"
+                    ></span>
+                </div>
+                <pci-project-data-processing-metrics-chart
+                    ng-if="$ctrl.metrics.activeTasks.data.length !== 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:73]:
```suggestion
                    data-ng-if="$ctrl.metrics.activeTasks.data.length !== 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 6]
```diff
@@ -33,61 +33,70 @@
...
+                >
+                </pci-project-data-processing-metrics-chart>
+                <!-- Tile item :: Blockmanager disk -->
+                <span class="information-tile__definition"
+                    >{{ :: 'data_processing_details_charts_blockmanager_disk' |
+                    translate }}</span
+                >
+                <div
+                    class="chart__nodata-container"
+                    ng-if="$ctrl.metrics.blockManagerDiskUsed.data.length === 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:86]:
```suggestion
                    data-ng-if="$ctrl.metrics.blockManagerDiskUsed.data.length === 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 7]
```diff
@@ -33,61 +33,70 @@
...
+                >
+                    <span
+                        data-translate="data_processing_details_no_data"
+                    ></span>
+                </div>
+                <pci-project-data-processing-metrics-chart
+                    ng-if="$ctrl.metrics.blockManagerDiskUsed.data.length !== 0"
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:93]:
```suggestion
                    data-ng-if="$ctrl.metrics.blockManagerDiskUsed.data.length !== 0"
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html]: Added here: https://github.com/ovh/manager/pull/7115/commits/4803e9a0fbcc662e37c712ba1b2807ee6f5dbb86

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 8]
```diff
@@ -33,61 +33,70 @@
...
+                    data-data="$ctrl.metrics.totalMemory.data"
+                    data-labels="$ctrl.metrics.totalMemory.labels"
+                    data-unit="Mi"
+                >
+                </pci-project-data-processing-metrics-chart>
+                <!-- Tile item :: Active tasks -->
+                <span class="information-tile__definition"
+                    >{{ :: 'data_processing_details_charts_num_active_tasks' |
+                    translate }}</span
+                >
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:63]:
```suggestion
                <span 
                         class="information-tile__definition"
                         data-translate="data_processing_details_charts_num_active_tasks"
                  >
                  </span>
```

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html [hunk 9]
```diff
@@ -33,61 +33,70 @@
...
+                    class="chart__container"
+                    data-data="$ctrl.metrics.activeTasks.data"
+                    data-labels="$ctrl.metrics.activeTasks.labels"
+                >
+                </pci-project-data-processing-metrics-chart>
+                <!-- Tile item :: Blockmanager disk -->
+                <span class="information-tile__definition"
+                    >{{ :: 'data_processing_details_charts_blockmanager_disk' |
+                    translate }}</span
+                >
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html:83]: Same as https://github.com/ovh/manager/pull/7115/files#r935548069

**Review by anooparveti — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html (hunk 8 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html]: Done here: https://github.com/ovh/manager/pull/7115/commits/a528dd8296a3c0c290385c37d694d0706c822e03

File: packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html (hunk 9 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/job-details/job-dashboard/job-dashboard.html]: Done here: https://github.com/ovh/manager/pull/7115/commits/a528dd8296a3c0c290385c37d694d0706c822e03

**Pull request merged**